content/docs/tasks.md

```diff
@@ -0,0 +1,178 @@
+
+**Note:** current implementation will apply all resources in the given order so that e.g. a Pod can mount previously created ConfigMap. However, this is not part of the specification and might change in the future (all resources can be applied concurrently). If you need [happens-before-guarantee](https://en.wikipedia.org/wiki/Happened-before) between your resources, use e.g. multiple serial steps.
+
+## Delete-Task
+
+A delete-task looks very similar to an apply-task, however, it will delete resources instead of creating them. Let's create a task that will uninstall the app we created above:
+
+```yaml
+tasks:
+  - name: remove
+    kind: Delete
+    spec:
+      resources:
+        - deployment.yaml
+```
+
+**Note:** deleting non-existing resources is always successful. As of version 0.9.0, KUDO will not wait for the resource to be actually removed and will finish the task when the API server accepts the deletion request. So in case of Pods, k8s imposes a default graceful termination period of 30 seconds, however, a delete-task will be done before that. KUDO 0.9.0 will not wait for [resource finalizers](https://kubernetes.io/docs/concepts/workloads/controllers/garbage-collection/) should they exist.
+
+## Pipe-Task
+
+Developing complicated operators often require generating files in one step and reusing them in a later one. A common example is generating custom certificates/dynamic configuration files in the bootstrap step and using them in the later deployment step of the service. This is were pipe-tasks can help: you can generate files in one task and save them either as a [Secret](https://cloud.google.com/kubernetes-engine/docs/concepts/secret) or a [ConfigMap](https://cloud.google.com/kubernetes-engine/docs/concepts/configmap) for use in subsequent steps. Let's see it in action. We will extend [first-operator](first-operator.md#your-first-kudo-operator) to generate a custom `index.html` page and deploy the Nginx server with it. Let's define a new pipe-task called `genwww`:
+
+```yaml
+tasks:
+  - name: genwww
+    kind: Pipe
+    spec:
+      pod: pipe-pod.yaml
+      pipe:
+        - file: /tmp/index.html
+          kind: ConfigMap
+          key: indexHtml
+```
+
+Pipe-task spec has two fields: a `pod` spec that will generate our `index.html` and a `pipe` spec that defines a list of files that will be persisted. Here we'll generate a `/tmp/index.html` file that we'll save as a `ConfigMap`. It can be referenced within template resources with the key `indexHtml` (more about that below). Pipe-task `spec.pod` field must reference a [core/v1 Pod](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.10/#pod-v1-core) template. However, there are limitations. Reasons for that are explained in the [corresponding KEP](https://github.com/kudobuilder/kudo/blob/master/keps/0017-pipe-tasks.md). In a nutshell:
```

Comment:
Is the config map removed afterwards? As config maps keep everything in plain text, is this currently really a good mechanism for secrets like custom certificates?

Reply:
For certificates use pipe `kind: Secret`